feat: Ensure postgres conf.d directory is set up

hunleyd opened the pull request:
Ensure the `/etc/postgresql-custom/conf.d` directory is correctly created and managed by the entrypoint script in Docker images for PostgreSQL 15, 17, and orioledb-17.

This change adds logic to `docker-entrypoint.sh` to:
- Create the `/etc/postgresql-custom/conf.d` directory if it does not exist.
- Set the ownership to `postgres:postgres`.
- Set the permissions to `0755`.

This ensures that custom PostgreSQL configuration files placed in this directory are correctly recognized and loaded by the PostgreSQL server during container startup, improving configuration management and extension support.

<img width="986" height="1281" alt="Screenshot 2025-12-03 at 11 05 31" src="https://github.com/user-attachments/assets/c9d3357c-1116-4037-9738-7594705db304" />

hunleyd committed:
feat: Ensure postgres conf.d directory is set up

Ensure the `/etc/postgresql-custom/conf.d` directory is correctly created and managed by the entrypoint script in Docker images for PostgreSQL 15, 17, and orioledb-17.

This change adds logic to `docker-entrypoint.sh` to:
- Create the `/etc/postgresql-custom/conf.d` directory if it does not exist.
- Set the ownership to `postgres:postgres`.
- Set the permissions to `0755`.

This ensures that custom PostgreSQL configuration files placed in this directory are correctly recognized and loaded by the PostgreSQL server during container startup, improving configuration management and extension support.

samrose reviewed (changes requested):
So far when I try to build Dockerfile-17 we're seeing 

```
docker build -f Dockerfile-17 -t supabase-postgres-17 .
[+] Building 1.5s (4/4) FINISHED                                                                                                                      docker:orbstack
 => [internal] load build definition from Dockerfile-17                                                                                                          0.0s
 => => transferring dockerfile: 12.18kB                                                                                                                          0.0s
 => resolve image config for docker-image://docker.io/docker/dockerfile:1.6                                                                                      0.7s
 => [auth] docker/dockerfile:pull token for registry-1.docker.io                                                                                                 0.0s
 => CACHED docker-image://docker.io/docker/dockerfile:1.6@sha256:ac85f380a63b13dfcefa89046420e1781752bab202122f8f50032edf31be0021                                0.0s
Dockerfile-17:257
--------------------
 256 |     RUN sed -i '/03775/a\\
 257 | >>> mkdir -p /etc/postgresql-custom/conf.d\
 258 | >>> chown postgres:postgres /etc/postgresql-custom/conf.d\
 259 | >>> chmod 0755 /etc/postgresql-custom/conf.d' /usr/local/bin/docker-entrypoint.sh
 260 |     
--------------------
ERROR: failed to build: failed to solve: dockerfile parse error on line 257: unknown instruction: mkdir
```

the problem is these commands need to be prefixed with `RUN`

samrose commented:
```
docker build -f Dockerfile-15 -t supabase-postgres-15 .
[+] Building 0.6s (3/3) FINISHED                                                                                                                      docker:orbstack
 => [internal] load build definition from Dockerfile-15                                                                                                          0.0s
 => => transferring dockerfile: 11.50kB                                                                                                                          0.0s
 => resolve image config for docker-image://docker.io/docker/dockerfile:1.6                                                                                      0.3s
 => CACHED docker-image://docker.io/docker/dockerfile:1.6@sha256:ac85f380a63b13dfcefa89046420e1781752bab202122f8f50032edf31be0021                                0.0s
Dockerfile-15:245
--------------------
 244 |     RUN sed -i '/03775/a\\
 245 | >>> mkdir -p /etc/postgresql-custom/conf.d\
 246 | >>> chown postgres:postgres /etc/postgresql-custom/conf.d\
 247 | >>> chmod 0755 /etc/postgresql-custom/conf.d' /usr/local/bin/docker-entrypoint.sh
 248 |     
--------------------
ERROR: failed to build: failed to solve: dockerfile parse error on line 245: unknown instruction: mkdir
```

samrose commented:
```
docker build -f Dockerfile-orioledb-17 -t supabase-postgres-orioledb . 

[+] Building 0.5s (3/3) FINISHED                                                                                                                      docker:orbstack
 => [internal] load build definition from Dockerfile-orioledb-17                                                                                                 0.0s
 => => transferring dockerfile: 13.06kB                                                                                                                          0.0s
 => resolve image config for docker-image://docker.io/docker/dockerfile:1.6                                                                                      0.2s
 => CACHED docker-image://docker.io/docker/dockerfile:1.6@sha256:ac85f380a63b13dfcefa89046420e1781752bab202122f8f50032edf31be0021                                0.0s
Dockerfile-orioledb-17:268
--------------------
 267 |     RUN sed -i '/03775/a\\
 268 | >>> mkdir -p /etc/postgresql-custom/conf.d\
 269 | >>> chown postgres:postgres /etc/postgresql-custom/conf.d\
 270 | >>> chmod 0755 /etc/postgresql-custom/conf.d' /usr/local/bin/docker-entrypoint.sh
 271 |     
--------------------
ERROR: failed to build: failed to solve: dockerfile parse error on line 268: unknown instruction: mkdir
```

samrose commented on Dockerfile-15:
```diff
@@ -241,6 +241,11 @@ ADD --chmod=0755 \
     https://github.com/docker-library/postgres/raw/889f9447cd2dfe21cccfbe9bb7945e3b037e02d8/15/bullseye/docker-entrypoint.sh \
     /usr/local/bin/docker-entrypoint.sh
 
+RUN sed -i '/03775/a\\
```
```suggestion
RUN sed -i '/03775/a\mkdir -p /etc/postgresql-custom/conf.d && 
         chown postgres:postgres /etc/postgresql-custom/conf.d && 
         chmod 0755 /etc/postgresql-custom/conf.d' /usr/local/bin/docker-entrypoint.sh
```

hunleyd committed:
build: Update Nix install command and Dockerfile configs

- Replaced `nix profile install .#<package>` with `nix profile add path:.#<package>` for installing Nix packages, which is both the recommended command, and allows nix to work with git worktrees when inside Docker.
- Ensured the `conf.d` directory for PostgreSQL configuration is correctly copied.
- Removed redundant `mkdir` and `chown` commands from the `docker-entrypoint.sh` script, as these are now handled by the `COPY` instruction.

hunleyd committed:
Merge branch 'develop' into INDATA-264

hunleyd committed:
Merge branch 'develop' into INDATA-264
